sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryUtils.java

```diff
@@ -133,23 +137,33 @@ static void addAttribute(Span span, String key, Object value, OpenTelemetrySchem
      * Parses an OpenTelemetry Status from AMQP Error Condition.
      *
      * @param span the span to set the status for.
-     * @param statusMessage description for this error condition.
+     * @param statusMessage Description for this error condition. If {@param throwable} is null, then the status
+     *     message is examined. If message is {@code null} then span is returned unmodified. Otherwise, if message is:
+     *     <ul>
+     *         <li>{@code "error"} then {@link StatusCode#ERROR} is set.</li>
+     *         <li>{@code "success"} then {@link StatusCode#OK} is set.</li>
+     *         <li>Otherwise, {@link StatusCode#ERROR} is set.</li>
+     *     </ul>
      * @param throwable the error occurred during response transmission (optional).
+     *
      * @return the corresponding OpenTelemetry {@link Span}.
      */
-    public static Span setError(Span span, String statusMessage, Throwable throwable) {
+    static Span setError(Span span, String statusMessage, Throwable throwable) {
         if (throwable != null) {
             span.recordException(throwable);
             return span.setStatus(StatusCode.ERROR, statusMessage);
         }
 
-        if (statusMessage != null) {
-            if ("error".equals(statusMessage)) {
-                return span.setStatus(StatusCode.ERROR);
-            }
-            return span.setStatus(StatusCode.ERROR, statusMessage);
+        if (statusMessage == null) {
+            return span;
         }
 
-        return span;
+        if (ERROR_STATUS_MESSAGE.equals(statusMessage)) {
+            return span.setStatus(StatusCode.ERROR);
+        } else if (SUCCESS_STATUS_MESSAGE.equals(statusMessage)) {
```

Comment:
@lmolkova Can you double check this is what you intended? I noticed in [EventHubsTracer](https://github.com/Azure/azure-sdk-for-java/pull/33559/files#diff-08ffb243e2d2c329044412fb18e940a70213e0a2d8fb5c814e25008485147f25L84) that it was passing in "success" as the default value, which in the old code would be a StatusCode.Error.  The javadocs for [OpenTelemetryTracer.end]([](https://github.com/Azure/azure-sdk-for-java/pull/33559/files#diff-08ffb243e2d2c329044412fb18e940a70213e0a2d8fb5c814e25008485147f25L84)) didn't specify this case though, so I wasn't sure my fix is what you wanted.

Reply:
not really, I got rid of "success" altogether here https://github.com/Azure/azure-sdk-for-java/pull/33600. along with other changes